sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2065,6 +2065,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val OPTIMIZER_HINTS_ENABLED =
+    buildConf("spark.sql.optimizer.hints.enabled")
```

Comment:
Can we have more direct names like `OPTIMIZER_IGNORE_HINTS`? Maybe, `spark.sql.optimizer.ignoreHints.enabled` like `spark.files.ignoreMissingFiles` or `spark.files.ignoreCorruptFiles`?

Reply:
@dongjoon-hyun Thank you. I have made the change.

Reply:
a little confused, why this config named `optimizer`, does it used in `analyze` ?

Reply:
@ulysses-you To the  user, its an optimization we are disabling. Thats why the "optimizer" name in the config.
Implementation wise, we handle it by keeping the nodes unresolved.  But semantic wise we are disabling an optimization.